app/imports/ui/components/EditProfile.jsx

```diff
@@ -0,0 +1,118 @@
+import LongTextField from 'uniforms-semantic/LongTextField';
+import SubmitField from 'uniforms-semantic/SubmitField';
+import HiddenField from 'uniforms-semantic/HiddenField';
+import ErrorsField from 'uniforms-semantic/ErrorsField';
+import PropTypes from 'prop-types';
+import { check } from 'meteor/check';
+import { Profiles, ProfileSchema } from '../../api/profile/profile';
+
+class EditProfile extends React.Component {
+  sites = [{ provider: 'linkedin', link: '' },
+            { provider: 'github', link: '' },
+            { provider: 'twitter', link: '' }];
+
+  socialSitesTemp = [];
+
+  state = { open: false };
+
+  /* Open modal */
+  show = () => this.setState({ open: true });
+
+  /* Close modal */
+  close = () => this.setState({ open: false });
+
+  /** On successful submit, insert the this.props.profile. */
+  submit(data) {
+    const { firstName, lastName, website, summary, image, zipCode, _id } = data;
+
+    /* Clear any social sites without any links */
+    let { socials } = data;
+    socials = socials.filter((social) => social.link);
+
+    /* Check's seem redundant but i included it anyways */
+    check(data, Object);
+    check(firstName, String);
+    check(lastName, String);
+
+    // Profiles.update(_id, { $unset: { socials } });
+    Profiles.update(_id, { $set: { firstName, lastName, socials, zipCode, summary, website, image } },
+        (error) => (error ?
+            Bert.alert({ type: 'danger', message: `Update failed: ${error.message}` }) :
+            Bert.alert({ type: 'success', message: 'Update succeeded' })));
+    this.close();
+  }
+
+  loadData = (profile) => () => {
+    let socialsIndex = 0;
+    this.socialSitesTemp = [];
+
+    /* Copy profile.socials.$.link to socialSitesTemp */
+    for (let i = 0; i < this.sites.length; i++) {
+      this.socialSitesTemp.push(Object.assign({}, this.sites[i]));
+      if (socialsIndex < profile.socials.length) {
+        if (profile.socials[socialsIndex].provider === this.socialSitesTemp[i].provider) {
+          this.socialSitesTemp[i].link = profile.socials[socialsIndex].link;
+          socialsIndex++;
+        }
+      }
+    }
+    Profiles.update(profile._id, { $set: { socials: this.socialSitesTemp } });
+    this.show();
+  };
+
+  /** Renders the Page for editing a single document. */
+  /** If the subscription(s) have been received, render the page, otherwise show a loading icon. */
+  render() {
+    const { open } = this.state;
+
+    return (
+        <div >
+          <Button color='green' content='Edit Profile' icon='edit'
+                  onClick={this.loadData(this.props.profile)}/>
+          <Modal open={open} onClose={this.close} closeIcon>
+            <Header icon='edit' content='Edit Profile'/>
+            <Modal.Content>
+              <Grid container centered>
+                <Grid.Column>
+                  <Header as="h2" textAlign="center">Edit Profile</Header>
+                  /* Need to bind this.submit call*/
```

Comment:
this is getting rendered on the page lol

Reply:
lol shit ill remove it